Add support for the Config API in the SchemaRegistry client

tjwp opened the pull request:
This change adds support for the [Config API](http://docs.confluent.io/3.1.2/schema-registry/docs/api.html#config).

Support for the config endpoints was also added to the `FakeSchemaRegistryServer`.

I also added a section to the README on using the `AvroTurf::SchemaRegistry` client directly, including the new config support.

dasch commented on README.md:
````diff
@@ -144,10 +147,17 @@ schema = <<-JSON
+registry.compatible?("person", schema)
+```
+
+The SchemaRegistry client can also change the global compatibility levelor the compatibility level for an individual subject using the [Config API](http://docs.confluent.io/3.1.2/schema-registry/docs/api.html#config):
````
Typo: `levelor`

dasch commented on lib/avro_turf/schema_registry.rb:
```diff
@@ -52,21 +52,48 @@ def check(subject, schema)
+  # Returns:
+  # - true if compatible
+  # - nil if the subject or version does not exist
+  # - false otherwise
```
> false if incompatible

tjwp commented:
That's interesting. I hadn't looked that closely at the single-object encoding since it's only in the unreleased spec on master. We're actually using this API with a [registry](https://github.com/salsify/avro-schema-registry) that we developed that is compatible with the Confluent API.

How would you feel about changing the name of the `SchemaRegistry`, `CachedSchemaRegistry`, and `FakeSchemaRegistryServer` classes, but also including an alias to the old name for each of those, and a deprecation notice?

Some people are obviously already using the `SchemaRegistry` API directly as evidenced by the addition of the `#compatible?` method, and issue/PR on where it is defined.

The aliases to the deprecated names could be removed at some point in the future when support for an Avro-spec compatible schema registry is added to AvroTurf. This approach avoids a compatibility break until some new functionality requires it.

dasch commented:
:+1:

tjwp commented:
I've made the change to rename `SchemaRegistry` and friends and added aliases for the old names.

I also added a Deprecation Notice to the README, a post-install message about the deprecation, and started a CHANGELOG listing the changes since the last release.

dasch commented:
Looks good! :+1:

tjwp commented:
I squashed and merged this. If you have time to cut a v0.8.0 release soon, that would be great!

dasch commented:
Done :+1:
